plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx

```diff
@@ -23,19 +24,198 @@ import { Icon, close } from '@wordpress/icons';
 /**
  * Internal dependencies
  */
-import type { BlockAttributes } from './types';
+import type { BlockAttributes, BlockVariationTriggerType } from './types';
+import { default as productFiltersIcon } from '../product-filters/icon';
+import { BlockOverlayAttribute as ProductFiltersBlockOverlayAttribute } from '../product-filters/types';
 import './editor.scss';
 
+const OverlayNavigationLabel = ( {
+	variation,
+}: {
+	variation: BlockVariationTriggerType;
+} ) => {
+	let label = __( 'Close', 'woocommerce' );
+	if ( variation === 'open-overlay' ) {
+		label = __( 'Filters', 'woocommerce' );
+	}
+
+	return <span>{ label }</span>;
+};
```

Comment:
If we create a component within another component, React will re-create those components every time the wrapper component re-renders. That way, React isn't able to perform any optimizations since it will consider those inner components as new components.

Reply:
I see, thank you!